flink-runtime/src/main/java/org/apache/flink/runtime/checkpoint/CheckpointCoordinator.java

```diff
@@ -318,6 +318,7 @@ public CheckpointCoordinator(
 			this.clock,
 			this.minPauseBetweenCheckpoints,
 			this.pendingCheckpoints::size,
+			() -> this.lastCheckpointCompletionRelativeTime,
```

Comment:
@rkhachatryan thanks for the fix.
After this change, the behavior is equal to 1.10 and would respect the min-pause setting now.

I'm wandering do we need to make the `lastCheckpointCompletionRelativeTime` and `pendingCheckpoints.size`  volatile.

Reply:
Thanks for reviewing @klion26.

`volatile` seems unnecessary because (both) values are written and read under the same lock.